src/encoded/views/download.py

```diff
@@ -0,0 +1,154 @@
+    ('File format', ('files.file_format')),
+    ('Size', ('files.file_size')),
+    ('Output type', ('files.output_type')),
+    ('Lab', ('files.lab')),
+    ('md5sum', ('files.md5sum')),
+    ('href', ('files.href')),
+    ('Assembly', ('files.assembly'))
+])
+
+
+def dict_generator(indict, pre=None):
+    pre = pre[:] if pre else []
+    if isinstance(indict, dict):
+        for key, value in indict.items():
+            if isinstance(value, dict):
+                for d in dict_generator(value, [key] + pre):
+                    yield d
+            elif isinstance(value, list) or isinstance(value, tuple):
+                for v in value:
+                    for d in dict_generator(v, [key] + pre):
+                        yield d
+            else:
+                yield pre + [key, value]
+    else:
+        yield indict
+
+
+@view_config(route_name='metadata', request_method='GET', renderer='tsv')
+def metadata_tsv(context, request):
+
+    param_list = parse_qs(request.matchdict['search_params'].encode('utf-8'))
+    param_list['field'] = []
+    header = []
+    file_attributes = []
+    for prop in _tsv_mapping:
+        header.append(prop)
+        param_list['field'] = param_list['field'] + [_tsv_mapping[prop]]
+        if _tsv_mapping[prop].startswith('files'):
+            file_attributes.append(_tsv_mapping[prop])
+    param_list['limit'] = ['all']
+    path = '/search/?%s' % urlencode(param_list, True)
+    results = embed(request, path, as_user=True)
+    rows = []
+    for row in results['@graph']:
+        if row['files']:
+            exp_data_row = []
+            new_data = []
+            for arr in dict_generator(row):
+                new_data.append(arr)
+            for column in header:
+                if not _tsv_mapping[column].startswith('files'):
+                    path = _tsv_mapping[column].split('.')
+                    temp = []
+                    for ld in new_data:
+                        if sorted(path) == sorted(ld[:-1]):
+                            if isinstance(ld[-1], list):
+                                temp = temp + ld[-1]
+                            else:
+                                if isinstance(ld[-1], bool):
+                                    temp.append(str(ld[-1]))
+                                else:
+                                    temp.append(str(ld[-1].encode('utf-8')))
+                    exp_data_row.append(', '.join(list(set(temp))))
+            for f in row['files']:
+                if 'files.file_format' in param_list:
+                    if f['file_format'] not in param_list['files.file_format']:
+                        continue
+                f['href'] = request.host_url + f['href']
+                for prop in file_attributes:
+                    value = ''
+                    if prop.split('.')[1] in f:
+                        value = f[prop.split('.')[1]]
+                    if prop.split('.')[1:][0] == 'accession':
```

Comment:
This will change. 
I have to rewrite this part to support path with more than 2 fragments.

eg: "files.platform.name"